yellowbrick/target/feature_correlation.py

```diff
@@ -0,0 +1,87 @@
+# yellowbrick.classifier.feature_correlation
+# Class balance visualizer for showing per-class support.
+#
+# Author    Zijie (ZJ) Poh <poh.zijie@gmail.com>
+# Author:   Benjamin Bengfort <bbengfort@districtdatalabs.com>
+# Created:  Wed Jul 29 15:30:40 2018 -0700
+#
+# ID: feature_correlation.py [5388065] neal@nhumphrey.com $
+
+"""
+Feature Correlation to Dependent Variable Visualizer.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import numpy as np
+
+from .base import TargetVisualizer
+
+from sklearn.feature_selection import (mutual_info_classif,
+                                       mutual_info_regression)
+from scipy.stats import pearsonr
+
+
+##########################################################################
+## Class Feature Correlation
+##########################################################################
+
+class FeatureCorrelation(TargetVisualizer):
+
+    method_func = {
+        'pearson': lambda X, y: [pearsonr(x, y)[0] for x in X.T],
+        'mutual_info': lambda X, y: mutual_info_regression(X, y),
+        'mutual_info_classif': lambda X, y: mutual_info_classif(X, y)
+    }
+    method_label = {
+        'pearson': 'Pearson Correlation',
+        'mutual_info': 'Mutual Information',
+        'mutual_info_classif': 'Mutual Information'
+    }
+
+    def __init__(self, ax=None, method='pearson', classification=False,
+                 **kwargs):
+        super(FeatureCorrelation, self).__init__(ax=None, **kwargs)
+
+        if classification and method == 'mutual_info':
+            method = 'mutual_info_classif'
+
+        # Parameters
+        self.set_params(
+            method=method
+        )
+
+    def fit(self, X, y, features_):
```

Comment:
Are features here the feature names? If so, it would be better to specify these in `__init__` rather than `fit`, and if the feature names aren't provided or discovered off of a `DataFrame` we can just identify them by index (e.g. 0, 1, 2, etc). I think the `FeatureImportances` visualizer had some mechanism for this right?

Reply:
That's right. If X is a `DataFrame`, then we can get the name from the `DataFrame` column names. If the feature names are not specify, we can just refer them by index.